classes/citation/Citation.php

```diff
@@ -20,99 +16,70 @@
 
 namespace PKP\citation;
 
-class Citation extends \PKP\core\DataObject
-{
-    /**
-     * Constructor.
-     *
-     * @param string $rawCitation an unparsed citation string
-     */
-    public function __construct($rawCitation = null)
-    {
-        parent::__construct();
-        $this->setRawCitation($rawCitation);
-    }
-
-    //
-    // Getters and Setters
-    //
-
-    /**
-     * Replace URLs through HTML links, if the citation does not already contain HTML links
-     *
-     * @return string
-     */
-    public function getCitationWithLinks()
-    {
-        $citation = $this->getRawCitation();
-        if (stripos($citation, '<a href=') === false) {
-            $citation = preg_replace_callback(
-                '#(http|https|ftp)://[\d\w\.-]+\.[\w\.]{2,6}[^\s\]\[\<\>]*/?#',
-                function ($matches) {
-                    $trailingDot = in_array($char = substr($matches[0], -1), ['.', ',']);
-                    $url = rtrim($matches[0], '.,');
-                    return "<a href=\"{$url}\">{$url}</a>" . ($trailingDot ? $char : '');
-                },
-                $citation
-            );
-        }
-        return $citation;
-    }
+use PKP\core\DataObject;
 
+class Citation extends DataObject
+{
     /**
-     * Get the rawCitation
-     *
-     * @return string
+     * Get the rawCitation.
      */
-    public function getRawCitation()
+    public function getRawCitation(): string|null
```

Comment:
I am wondering when it is the case where the rawCitation would not exist? :thinking: